submissions/Roophee/JS - DOM/index.js

```diff
@@ -45,23 +45,13 @@ const contentStorage = {
 
 let dynamicNav = function (e) {
     e.preventDefault();
-    let fragment = document.createDocumentFragment();
+    let insertNav = '';
 
     for (let i of Object.keys(contentStorage)) {
-        let navItem = document.createElement("li");
-        let navLink = document.createElement("a");
-        let linkText = document.createElement("span");
-
-        navItem.className = "nav__item";
-        navLink.className = "nav__link";
-        navLink.href = "#top";
-        linkText.innerText = i;
-
-        navLink.appendChild(linkText);
-        navItem.appendChild(navLink);
-        fragment.appendChild(navItem);
+        let insetNavItem = `<li class="nav__item"><a href="#top" class="nav__link"><span>${i}</span></a></li> \n`;
+        insertNav += insetNavItem;
     }
-    navListOfElements.appendChild(fragment);
+    navListOfElements.innerHTML = insertNav;
```

Comment:
`innerHTML` rewrite all HTML inside parent block. For this time I'll merge it but in the future avoid this approach.
I mean create wrapper using `document.createElement` or `document.createDocumentFragment` and inner part fill by template string with markup with all classes and attributes.
Then just append your filled wrapper to the corresponding place.)